### ISSUE TITLE
Av1an does not encode and quit

### ISSUE THREAD

**Issue by ahmadrr** [master-of-zen/Av1an#360]
I compiled Av1an from git just now, it doesn't encode and just quit, even with running as admin. The only thing that working is showing help.

d:\Video\av1\target\release>av1an -i d:\test.mp4 (https://user-images.githubusercontent.com/44171723/133711285-f71a1c58-791a-4f15-8fdb-263052d07248.mp4)

Running the command above returns nothing, no new file/folder created anywhere.

**Comment by redzic** [comment]
Where are you running av1an from, the default windows command prompt or something else? How did you compile it, did you use mingw or the microsoft build tools?

**Comment by ahmadrr** [comment]
I was running av1an from the default windows command prompt, not powershell or anything. I compiled it using the microsoft build tools by running cargo build --release

**Comment by ahmadrr** [comment]
![image](https://user-images.githubusercontent.com/44171723/133711675-1e0c2c2a-0c56-4bff-ab5f-74e5d1aeec4a.png)
![image](https://user-images.githubusercontent.com/44171723/133711686-614082bc-fae9-403b-9a67-2ab83f3e13f1.png)

**Comment by redzic** [comment]
For some reason, there seems to sometimes be problems with running av1an on a different drive other than the C drive on windows, and also with using clips that are not on the C drive, but I will have to look into it further. Can you try compiling from the C drive, and using a test clip from the C drive for now to at least determine that it's not some other issue?

**Comment by ahmadrr** [comment]
Still, nothing happened.

c:\av1>cargo build --release
   Compiling autocfg v1.0.1
   Compiling winapi v0.3.9
   Compiling libc v0.2.101
   Compiling proc-macro2 v1.0.29
   Compiling unicode-xid v0.2.2
   Compiling cfg-if v1.0.0
   Compiling syn v1.0.76
   Compiling memchr v2.4.1
   Compiling crossbeam-utils v0.8.5
   Compiling lazy_static v1.4.0
   Compiling bitflags v1.2.1
   Compiling crossbeam-epoch v0.9.5
   Compiling rayon-core v1.9.1
   Compiling version_check v0.9.3
   Compiling log v0.4.14
   Compiling jobserver v0.1.24
   Compiling ucd-trie v0.1.3
   Compiling regex-syntax v0.6.25
   Compiling scopeguard v1.1.0
   Compiling semver v1.0.4
   Compiling serde_derive v1.0.130
   Compiling rustversion v1.0.5
   Compiling either v1.6.1
   Compiling ryu v1.0.5
   Compiling byteorder v1.4.3
   Compiling adler32 v1.2.0
   Compiling serde v1.0.130
   Compiling crc32fast v1.2.1
   Compiling radium v0.5.3
   Compiling ntapi v0.3.6
   Compiling unicode-segmentation v1.8.0
   Compiling unicode-width v0.1.8
   Compiling adler v1.0.2
   Compiling lexical-core v0.7.6
   Compiling rust_hawktracer_proc_macro v0.4.1
   Compiling gimli v0.25.0
   Compiling bytes v1.1.0
   Compiling failure_derive v0.1.8
   Compiling color_quant v1.1.0
   Compiling rust_hawktracer_normal_macro v0.4.1
   Compiling strsim v0.8.0
   Compiling plotters-backend v0.3.2
   Compiling wyz v0.2.0
   Compiling funty v1.1.0
   Compiling doc-comment v0.3.3
   Compiling vapoursynth-sys v0.3.0
   Compiling vec_map v0.8.2
   Compiling weezl v0.1.5
   Compiling bytemuck v1.7.2
   Compiling tap v1.0.1
   Compiling static_assertions v1.1.0
   Compiling byte-slice-cast v1.0.0
   Compiling jpeg-decoder v0.1.22
   Compiling noop_proc_macro v0.3.0
   Compiling serde_json v1.0.67
   Compiling rustc-demangle v0.1.21
   Compiling arrayvec v0.5.2
   Compiling paste v1.0.5
   Compiling itoa v0.4.8
   Compiling bitstream-io v1.2.0
   Compiling anyhow v1.0.43
   Compiling sysinfo v0.19.2
   Compiling encode_unicode v0.3.6
   Compiling once_cell v1.8.0
   Compiling float-ord v0.2.0
   Compiling arrayvec v0.7.1
   Compiling cfg-if v0.1.10
   Compiling std_prelude v0.2.12
   Compiling number_prefix v0.4.0
   Compiling ttf-parser v0.12.3
   Compiling smawk v0.3.1
   Compiling yansi v0.5.0
   Compiling glob v0.3.0
   Compiling y4m v0.7.0
   Compiling pin-project-lite v0.2.7
   Compiling strsim v0.10.0
   Compiling splines v4.0.3
   Compiling shlex v1.1.0
   Compiling num-traits v0.2.14
   Compiling num-integer v0.1.44
   Compiling memoffset v0.6.4
   Compiling rayon v1.5.1
   Compiling num-bigint v0.3.3
   Compiling num-rational v0.3.2
   Compiling num-iter v0.1.42
   Compiling miniz_oxide v0.4.4
   Compiling tokio v1.11.0
   Compiling cc v1.0.70
   Compiling pest v2.1.3
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling nom v6.1.2
   Compiling itertools v0.10.1
   Compiling miniz_oxide v0.3.7
   Compiling deflate v0.8.6
   Compiling textwrap v0.11.0
   Compiling heck v0.3.3
   Compiling rust_hawktracer v0.7.0
   Compiling addr2line v0.16.0
   Compiling plotters-svg v0.3.1
   Compiling gif v0.11.2
   Compiling semver-parser v0.10.2
   Compiling backtrace v0.3.61
   Compiling num_cpus v1.13.0
   Compiling which v4.2.2
   Compiling quote v1.0.9
   Compiling aho-corasick v0.7.18
   Compiling object v0.26.2
   Compiling crossbeam-channel v0.5.1
   Compiling rustc_version v0.4.0
   Compiling err-derive v0.2.4
   Compiling semver v0.11.0
   Compiling png v0.16.8
   Compiling bitvec v0.19.5
   Compiling atty v0.2.14
   Compiling winapi-util v0.1.5
   Compiling wio v0.2.2
   Compiling time v0.1.44
   Compiling miow v0.3.7
   Compiling terminal_size v0.1.17
   Compiling ctrlc v3.2.0
   Compiling simd_helpers v0.1.0
   Compiling regex v1.5.4
   Compiling rustc_version v0.3.3
   Compiling clap v2.33.3
   Compiling same-file v1.0.6
   Compiling dwrote v0.11.0
   Compiling console v0.14.1
   Compiling mio v0.7.13
   Compiling crossbeam-deque v0.8.1
   Compiling chrono v0.4.19
   Compiling pathfinder_simd v0.5.1
   Compiling stfu8 v0.2.4
   Compiling synstructure v0.12.5
   Compiling unicode-linebreak v0.1.2
   Compiling walkdir v2.3.2
   Compiling thiserror-impl v1.0.28
   Compiling num-derive v0.3.3
   Compiling arg_enum_proc_macro v0.3.1
   Compiling strum_macros v0.21.1
   Compiling indicatif v0.17.0-beta.1
   Compiling vergen v3.2.0
   Compiling structopt-derive v0.4.16
   Compiling thiserror v1.0.28
   Compiling strum v0.21.0
   Compiling image v0.23.14
   Compiling failure v0.1.8
   Compiling av-bitstream v0.1.2
   Compiling av-data v0.3.0
   Compiling flexi_logger v0.18.1
   Compiling structopt v0.3.23
   Compiling nasm-rs v0.2.1
   Compiling path_abs v0.5.1
   Compiling dashmap v4.0.2
   Compiling v_frame v0.2.4
   Compiling pathfinder_geometry v0.5.1
   Compiling vapoursynth v0.3.0
   Compiling plotters-bitmap v0.3.1
   Compiling av-format v0.3.1
   Compiling textwrap v0.14.2
   Compiling rav1e v0.5.0-beta.2
   Compiling font-kit v0.10.1
   Compiling av-ivf v0.2.2
   Compiling plotters v0.3.1
   Compiling av-scenechange v0.7.0
   Compiling av1an-core v0.1.0 (C:\av1\av1an-core)
   Compiling av1an-cli v0.1.1 (C:\av1\av1an-cli)
   Compiling av1an v0.1.0 (C:\av1)
    Finished release [optimized] target(s) in 3m 15s

c:\av1>cd /d target\release

c:\av1\target\release>av1an -i c:\test.mp4

c:\av1\target\release>

**Comment by n00mkrad** [comment]
VapourSynth installed and in PATH?

**Comment by redzic** [comment]
vapoursynth is required to compile av1an, so it must be installed. I suspect it's some kind of compilation/linking related issue. I remember having a similar kind of issue and it was fixed with some kind of configuration related change.

Can you try compiling with msys2/mingw to see if that fixes things?

**Comment by redzic** [comment]
Also, can you try `av1an -i <input> -m hybrid`? If it works, then that means that there is some kind of problem with the dynamic linking of vapoursynth on your system

**Comment by ahmadrr** [comment]
VapourSynth installed and in PATH? Yes.

I uninstalled python 3.8 and 3.9, also deleted Python38 and Python39 folders from Program Files but left the VapourSynth folders, extracted core.msi from Python 3.8.10 and Python 3.9.7 installers, put python38.dll and python39.dll in C:\Windows\system32. Then I installed Python 3.8.10 and 3.9.7 with only "install to all users" and "precompile standard library" options checked, installed Vapoursynth R55 on Python 3.9. Next, I compiled Av1an from yesterday's master.zip in d:\videos\av1 and it can sucessfully* encode [test.mp4](https://user-images.githubusercontent.com/46526140/122739042-3ec02900-d28b-11eb-8ef0-2e5e3e846525.mp4)

**Comment by lefth** [comment]
I also had a segmentation fault at launch. It was also due to vapoursynth not initializing. I found the suggestion in the [vapoursynth docs](http://www.vapoursynth.com/doc/installation.html) to test by running `vspipe --version`. They also suggest solving this by registering the path where vapoursynth is installed (note the "x" which needs to be replaced):

    PYTHONPATH=/usr/local/lib/python3.x/site-packages

Setting that solved it. If the variable is already set, append to its value with `PYTHONPATH="$PYTHONPATH:/usr/local/lib/python3.x/site-packages"`
